Build fails because of unknown wget option `show-progress`

### Describe the bug

Building with kraftkit fails because GNU Wget2 2.1.0 doesn't accept the flag `show-progress`. It looks like this flag is only available for [GNU wget 1](https://man7.org/linux/man-pages/man1/wget.1.html). Based on the man page, the `--force-progress` flag for GNU Wget2 has a similar effect to the `--show-progress` flag. The change from `--show-progress` to `--force-progress` is mentioned on the [Wget2 gitlab page](https://gitlab.com/gnuwget/wget2/-/wikis/Home):

![image](https://github.com/unikraft/kraftkit/assets/89748807/49d98aa5-25b2-4d3b-adee-72e753251443)

### Steps to reproduce

```
git clone https://github.com/unikraft/catalog
cd catalog/library/nginx/1.25/
# Based on https://unikraft.org/guides/catalog-behind-the-scenes#nginx
docker run -d --name buildkitd --privileged moby/buildkit:latest
export KRAFTKIT_BUILDKIT_HOST=docker-container://buildkitd
kraft build --plat qemu --arch x86_64
```

### Expected behavior

The right flat for Wget should be chosen based on the version of Wget.

### Which architectures were you using or does this bug affect?

x86_64

### Which operating system were you using or does this bug affect?

linux/fedora

### Relevant log output

```shell
$ kraft build --plat qemu --arch x86_64
[+] updating index... done!                                                                              [2.6s]
[+] finding app/elfloader:staging... done!                                                               [0.4s]
[+] pulling app/elfloader:staging   ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• 100% [0.0s]
[+] finding core/unikraft:staging... done!                                                               [0.4s]
[+] finding lib/lwip:staging... done!                                                                    [0.3s]
[+] finding lib/libelf:staging... done!                                                                  [0.4s]
[+] pulling lib/libelf:staging      ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• 100% [0.0s]
[+] pulling lib/lwip:staging        ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• 100% [0.0s]
[+] pulling core/unikraft:staging   ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• 100% [0.0s]
 W  could not parse image name: invalid reference format: repository name must be lowercase
[+] building rootfs... done!                                                                     x86_64 [17.1s]
[+] configuring nginx (qemu/x86_64) ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• 100% [0.5s]
<!> building nginx (qemu/x86_64)                                                                      0% [0.6s]
 i    LN      Makefile                                                                                         
 i    GEN     libuklibid: libraries.in.new                                                                     
 i    CP      config                                                                                           
 W  Unknown option 'show-progress'                                                                             
 i    WGET    liblwip: https://github.com/unikraft/fork-lwip/archive/refs/heads/UNIKRAFT-2_1_x.zip             
 W  unzip:  cannot find or open /home/thasso/Work/catalog/library/nginx/1.25/.unikraft/build/liblwip/UNIKRAFT-2
 W  make[1]: *** [/home/thasso/Work/catalog/library/nginx/1.25/.unikraft/libs/lwip/Makefile.uk:60: /home/thasso
 i    UNZIP   liblwip: UNIKRAFT-2_1_x.zip                                                                      
 W  make[1]: *** Waiting for unfinished jobs....                                                               
 i    GEN     config.h                                                                                         
 W  make: *** [Makefile:1170: sub-make] Error 2                                                                
                                                                                                               
 E  could not complete build: build failed: exit status 2
```

--- a/Makefile
+++ b/Makefile
@@ -679,7 +679,11 @@ GZIP		:= gzip
 TAR		:= tar
 UNZIP		:= unzip -qq -u
 GIT		:= git
-WGET		:= wget
+ifneq (,$(findstring Wget2,$(shell wget --version)))
+WGET := wget -q --force-progress --progress=bar
+else
+WGET := wget -q --show-progress --progress=bar
+endif
 PYTHON          := python3
 SHA1SUM		:= sha1sum -b
 SHA256SUM	:= sha256sum -b

--- a/support/build/Makefile.rules
+++ b/support/build/Makefile.rules
@@ -527,7 +527,7 @@ endef
 define fetchas =
 $(BUILD_DIR)/$(1)/$(3):
 	$(call verbose_cmd,WGET,$(1)':' $(2), \
-	 $(WGET) -q --show-progress --progress=bar -O $(BUILD_DIR)/$(1)/$(3) $(2) || \
+	 $(WGET) -O $(BUILD_DIR)/$(1)/$(3) $(2) || \
 	 $(RM) $(BUILD_DIR)/$(1)/$(3))
 
 $(call _chksum_origin,$(1),$(BUILD_DIR)/$(1)/$(3),$(BUILD_DIR)/$(1)/.chksum)
@@ -541,8 +541,8 @@ endef
 define fetchas2 =
 $(BUILD_DIR)/$(1)/$(4):
 	$(call verbose_cmd,WGET,$(1)':' $(2) [retry-with: $(3)], \
-	 $(WGET) -q --show-progress --progress=bar -O $(BUILD_DIR)/$(1)/$(4) $(2) || \
-	 $(WGET) -q --show-progress --progress=bar -O $(BUILD_DIR)/$(1)/$(4) $(3) || \
+	 $(WGET) -O $(BUILD_DIR)/$(1)/$(4) $(2) || \
+	 $(WGET) -O $(BUILD_DIR)/$(1)/$(4) $(3) || \
 	 $(RM) $(BUILD_DIR)/$(1)/$(4))
 
 $(call _chksum_origin,$(1),$(BUILD_DIR)/$(1)/$(4),$(BUILD_DIR)/$(1)/.chksum)

